warehouse/scripts/run_and_upload.py

```diff
@@ -55,24 +55,23 @@ def get_command(*args) -> List[str]:
 
     subprocess.run(get_command("compile")).check_returncode()
```

Comment:
so just to confirm, if compile fails, we don't do anything else? but if compile succeeds we will now attempt: run; docs deploy; metabase sync; freshness check (if configured) and will raise failures from all those steps at the end?

Reply:
There's no real logic change here except checking that the docs did deploy at the end. We've always checked for compile first, since a project that can't compile can't do anything else.

Reply:
yeah sorry, I get that this isn't changing much here I mostly just wanted to make sure I understood the overall order of operations. and specifically that we will attempt ALL steps even if the dbt run has errors (which I think we *weren't* doing for a while before?)

Reply:
Ah yeah that was true originally and was wrong. A partial/errored run will still be reflected in the production database so we should be updating the docs/Metabase. We don't really have the bandwidth to do blue/green deploys.